sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -842,3 +699,302 @@ object HyperLogLogPlusPlus {
+        0.0
+      }
+
+      buffer.setDouble(mutableAggBufferOffset, updateM0)
+      buffer.setDouble(mutableAggBufferOffset + 1, updateM1)
+      buffer.setDouble(mutableAggBufferOffset + 2, updateM2)
+      buffer.setDouble(mutableAggBufferOffset + 3, updateM3)
+      buffer.setDouble(mutableAggBufferOffset + 4, updateM4)
+    }
+  }
+
+  /** Merge two central moment buffers. */
+  override def merge(buffer1: MutableRow, buffer2: InternalRow): Unit = {
+    val zeroMoment1 = buffer1.getDouble(mutableAggBufferOffset)
+    val zeroMoment2 = buffer2.getDouble(inputAggBufferOffset)
+    val firstMoment1 = buffer1.getDouble(mutableAggBufferOffset + 1)
+    val firstMoment2 = buffer2.getDouble(inputAggBufferOffset + 1)
+    val secondMoment1 = buffer1.getDouble(mutableAggBufferOffset + 2)
+    val secondMoment2 = buffer2.getDouble(inputAggBufferOffset + 2)
+    val thirdMoment1 = buffer1.getDouble(mutableAggBufferOffset + 3)
+    val thirdMoment2 = buffer2.getDouble(inputAggBufferOffset + 3)
+    val fourthMoment1 = buffer1.getDouble(mutableAggBufferOffset + 4)
+    val fourthMoment2 = buffer2.getDouble(inputAggBufferOffset + 4)
+
+    val zeroMoment = zeroMoment1 + zeroMoment2
+    val delta = firstMoment2 - firstMoment1
+    val deltaN = delta / zeroMoment
+
+    val firstMoment = firstMoment1 + deltaN * zeroMoment2
+
+    val secondMoment = if (maxMoment >= 2) {
+      secondMoment1 + secondMoment2 + delta * deltaN * zeroMoment1 * zeroMoment2
+    } else {
+      0.0
+    }
+
+    val thirdMoment = if (maxMoment >= 3) {
+      thirdMoment1 + thirdMoment2 + deltaN * deltaN * delta * zeroMoment1 * zeroMoment2 *
+        (zeroMoment1 - zeroMoment2) + 3.0 * deltaN *
+        (zeroMoment1 * secondMoment2 - zeroMoment2 * secondMoment1)
+    } else {
+      0.0
+    }
+
+    val fourthMoment = if (maxMoment >= 4) {
+      fourthMoment1 + fourthMoment2 + deltaN * deltaN * deltaN * delta * zeroMoment1 *
+        zeroMoment2 * (zeroMoment1 * zeroMoment1 - zeroMoment1 * zeroMoment2 +
+          zeroMoment2 * zeroMoment2) + deltaN * deltaN * 6.0 *
+        (zeroMoment1 * zeroMoment1 * secondMoment2 + zeroMoment2 * zeroMoment2 * secondMoment1) +
+        4.0 * deltaN * (zeroMoment1 * thirdMoment2 - zeroMoment2 * thirdMoment1)
+    } else {
+      0.0
+    }
+
+    buffer1.setDouble(mutableAggBufferOffset, zeroMoment)
+    buffer1.setDouble(mutableAggBufferOffset + 1, firstMoment)
+    buffer1.setDouble(mutableAggBufferOffset + 2, secondMoment)
+    buffer1.setDouble(mutableAggBufferOffset + 3, thirdMoment)
+    buffer1.setDouble(mutableAggBufferOffset + 4, fourthMoment)
+  }
+}
+
+case class Stddev(child: Expression) extends CentralMomentAgg(child) {
+
+  override def prettyName: String = "stddev"
+
+  protected val maxMoment = 2
+
+  def eval(buffer: InternalRow): Any = {
+    // stddev = sqrt(M2 / (M0 - 1))
+    val M0 = buffer.getDouble(mutableAggBufferOffset)
+    val M2 = buffer.getDouble(mutableAggBufferOffset + 2)
+
+    if (M0 == 0.0) {
+      0.0
```

Comment:
R and SciPy differ on this. R outputs NaN in divide by zero situations where SciPy outputs 0. I will assume we stick with R implementation.